packages/web3-core/types/index.d.ts

```diff
@@ -410,9 +413,14 @@ export interface LogsOptions {
 
 export type BlockNumber = string | number | BN | BigNumber | 'latest' | 'pending' | 'earliest' | 'genesis';
 
+export interface AbstractProvider extends EventEmitter {
+    send(payload: JsonRpcPayload, callback: (error: Error | null, result?: JsonRpcResponse) => void): void;
```

Comment:
The interface may also need to require sendAsync function for a library that only supports a legacy provider.

e.g. https://github.com/MetaMask/eth-json-rpc-middleware/blob/9464aa2085c63b43f1e5ed569bd08b0697c53d39/providerAsMiddleware.js#L6

Reply:
Hey @odanado thanks again for this PR and sorry for the delay, finally getting back to it :)

I think adding `sendAsync` would be a great idea, do you mind adding it? [ethers.js/web3-provider](https://github.com/ethers-io/ethers.js/blob/56af4413b1dd1787db68985e0b612b63d86fdf7c/packages/providers/src.ts/web3-provider.ts#L17) defines it like so:

```typescript
sendAsync?: (request: { method: string, params?: Array<any> }, callback: (error: any, response: any) => void) => void
```

I'll add the latest EIP-1193 compatibility in a future PR since we just moved the document into Last Call with a 6-week review period and MetaMask should be deploying their implementation soon.

Reply:
@ryanio Thank you for the comment.
I updated.

- remove EventEmitter
- add sendAsync method